src/SimplePromiseQueue.ts

```diff
@@ -1,18 +1,32 @@
 // tslint:disable: variable-name
 export default class SimplePromiseQueue {
+  private readonly _skipIntermediateTasks: boolean
   private readonly _queue: Array<() => Promise<void>> = []
   private _runner: Promise<void> | null = null
 
+  public constructor(skipIntermediateSaves: boolean) {
+    this._skipIntermediateTasks = skipIntermediateSaves
+  }
+
   public enqueue(task: () => Promise<void>) {
     this._queue.push(task)
     this.flushQueue()
   }
 
   public async flushQueue(): Promise<void> {
     const runner = async () => {
-      while (this._queue.length) {
-        const nextTask = this._queue.shift()!
-        await nextTask()
+      if (this._skipIntermediateTasks) {
+        // With this option enabled, we only execute the last task in the queue
+        while (this._queue.length) {
+          const nextTask = this._queue[this._queue.length - 1]
+          this._queue.splice(0, this._queue.length)
+          await nextTask()
+        }
+      } else {
+        while (this._queue.length) {
+          const nextTask = this._queue.shift()!
+          await nextTask()
+        }
```

Comment:
Instead of this should  enqueue just keep track of the latest item?

Reply:
That makes sense. I implemented this as a new class.